server.js

```diff
@@ -22,9 +14,125 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+//ROUTES
+
+//Homepage
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Possible routes: /boardgames/:id, /year/:year, /rating/:rating (change :rating to "average" or "rank" ')
+})
+
+//Specific boardgames, requires id
+app.get("/boardgames/:id", (req, res) => {
+  const id = req.params.id
+  const boardgame = boardgamesData.find((game) => game.id === +id)
+  if (boardgame) {
+    res.json(boardgame)
+  } else {
+    res.status(404).send(`No game found with id: ${id}`)
+  }
+})
+
+//Pages or search for name
+app.get("/boardgames", (req, res) => {
+  //Query for name
+  const nameSearch = req.query.name
+
+  //Query for page
+  //Makes the number written in url to integer
+  let pageSearch = parseInt(req.query.page)
+
+  //Checks how many pages there is if every page has 10 objects
+  const pageCount = Math.ceil(boardgamesData.length / 10)
+
+  //If there's no page-query in the url, show first page
+  if (!pageSearch) {
+    pageSearch = 1
+  }
+  //If the page-query is bigger than the pageCount it should show the last page
+  else if (pageSearch > pageCount) {
+    pageSearch = pageCount
+  }
+
+  //If a nameSearch, filter and return only the ones that includes the query
+  if (nameSearch) {
+    const filteredSearch = boardgamesData.filter(game => {
+      const gameName = game.name.toString()
+      return gameName.toLowerCase().includes(nameSearch.toLowerCase())
+    })
+    const pageCount = Math.ceil(filteredSearch.length / 10)
+    if (!pageSearch) {
+      pageSearch = 1
+    }
+    else if (pageSearch > pageCount) {
+      pageSearch = pageCount
+    }
+
+    if (filteredSearch.length === 0) {
+      res.status(404).send(`Couldn't find any boardgame with "${nameSearch}" in the name`)
+    } else {
+      res.json(filteredSearch.slice(pageSearch * 10 - 10, pageSearch * 10))
+    }
+  }
+
+  //Slice the data, begin on the page written in query
+  res.json(boardgamesData.slice(pageSearch * 10 - 10, pageSearch * 10))
+})
+
+//Boardgame from year, requires year
+app.get("/year/:year", (req, res) => {
```

Comment:
It would be nice if this and the rating endpoints were all part of the `/boardgames` endpoint (and it'd be more RESTful).  One way to approach having multiple filters in one endpoint is to make a copy of the array of data and then slowly go through re-assigning it when there's some filter.  Something along these lines:

```js
let data = boardgamesData

if (nameSearch) {
  data = data.filter(...)
}

if (req.params.year) {
  data = data.filter(...)
}

if (req.params.rating) {
  data = data.filter(...)
}

res.json(data)
```

So you keep filtering and re-assigning the variable before finally returning it once all the possible filters have been run. Does that make sense?